automated/android/noninteractive-tradefed/tradefed.sh

```diff
@@ -82,31 +87,33 @@ if [ -e "${TEST_PATH}/testcases/vts/testcases/kernel/linux_kselftest/kselftest_c
     sed -i "/suspend/d" "${TEST_PATH}"/testcases/vts/testcases/kernel/linux_kselftest/kselftest_config.py
 fi
 
-if [ -n "${AP_SSID}" ] && [ -n "${AP_KEY}" ]; then
-    # try to connect to wifi with the feature provided by tradefed by default
-    # when AP_SSID and AP_KEY are specified for this tradefed test action explicitly
-    TEST_PARAMS="${TEST_PARAMS} --wifi-ssid ${AP_SSID} --wifi-psk ${AP_KEY}"
-else
-    # try to connect to wifi with the external AdbJoinWifi apk from
-    # https://github.com/steinwurf/adb-join-wifi
-    # if AP_SSID and AP_KEY are not specified for this tradefed test action
-    adb_join_wifi "${AP_SSID}" "${AP_KEY}"
-fi
+if [ "X${INTERNET_ACCESS}" = "Xtrue" ]; then
+    if [ -n "${AP_SSID}" ] && [ -n "${AP_KEY}" ]; then
+        # try to connect to wifi with the feature provided by tradefed by default
+        # when AP_SSID and AP_KEY are specified for this tradefed test action explicitly
+        TEST_PARAMS="${TEST_PARAMS} --wifi-ssid ${AP_SSID} --wifi-psk ${AP_KEY}"
+    else
+        # try to connect to wifi with the external AdbJoinWifi apk from
+        # https://github.com/steinwurf/adb-join-wifi
+        # if AP_SSID and AP_KEY are not specified for this tradefed test action
+        adb_join_wifi "${AP_SSID}" "${AP_KEY}"
+    fi
 
-# wait for a while till the wifi connecting operation finished
-sleep 60
+    # wait for a while till the wifi connecting operation finished
+    sleep 60
 
-SERVER="www.google.com"
-info_msg "device-${ANDROID_SERIAL}: About to check with ping ${SERVER}..."
-if adb shell 'ping -c 10 '"${SERVER}"'; echo exitcode: $?' | grep -q "exitcode: 0"; then
-    report_pass "network-available"
-else
-    report_fail "network-available"
-    # print more debug information
-    adb shell ip address || true
-    adb shell ping -c 10 8.8.8.8 || true
-    # to be caught by the yaml file
-    exit 100
+    SERVER="www.google.com"
+    info_msg "device-${ANDROID_SERIAL}: About to check with ping ${SERVER}..."
+    if adb shell 'ping -c 10 '"${SERVER}"'; echo exitcode: $?' | grep -q "exitcode: 0"; then
+        report_pass "network-available"
+    else
+        report_fail "network-available"
+        # print more debug information
+        adb shell ip address || true
+        adb shell ping -c 10 8.8.8.8 || true
+        # to be caught by the yaml file
+        exit 100
+    fi
```

Comment:
Maybe it would make sense here to have "BACKUP_SERVER" variable? 

BACKUP_SERVER="8.8.8.8"

and then

if adb shell "ping -c 5 ${SERVER}" > /dev/null 2>&1; then
....
else
    if adb shell "ping -c 5 ${BACKUP_SERVER}" > /dev/null 2>&1; then
...
else
<debug info>

Reply:
ping 8.8.8.8 is to check if there is any DNS issue in the lab. When ping www.google.com failed, but ping 8.8.8.8 passed, then it means there is some DNS issue in the lab, but it's still the network failure.

Reply:
Thanks Yongqin. Sorry wasn't clear, I meant to use DNS_TEST_SERVER or BACKUP_SERVER as a variable at the top. I would also add some additional debug info in, when it does fail. Something like this:

```
SERVER="www.google.com"
DNS_TEST_SERVER="8.8.8.8"

info_msg "device-${ANDROID_SERIAL}: Checking network connectivity by pinging ${SERVER}..."

# Try pinging www.google.com first
if adb shell "ping -c 10 ${SERVER}" > /dev/null 2>&1; then
    report_pass "network-available"
else
    info_msg "Ping to ${SERVER} failed, testing DNS by pinging ${DNS_TEST_SERVER}..."

    # If ping to 8.8.8.8 succeeds, it's likely a DNS issue
    if adb shell "ping -c 10 ${DNS_TEST_SERVER}" > /dev/null 2>&1; then
        report_fail "network-available: DNS issue detected"
        
        # DNS-specific debug information
        info_msg "DNS resolution issue suspected; gathering DNS configuration information..."
        adb shell getprop | grep dns || true  # Display DNS settings

    else
        report_fail "network-available: general network failure"
        
        # General network debug information
        info_msg "Network connectivity issue detected; gathering debug information..."
        adb shell ip address || true
        adb shell route || true
        adb shell ifconfig || true
    fi
    
    # Exit with error to trigger YAML file handling
    exit 100
fi

```

Reply:
thanks, that looks really good, let's submit it as another PR instead of being mixed here.